### PR TITLE
[DOCS] Update TLS encryption instructions

### PR DESCRIPTION
**Pull request by alaudazzi**
This PR proposes the following changes:

- Move the **Encrypting communications** section after the Overview.
- Add a warning to clarify that general knowledge about TLS and SSL is expected.
- In the **Setting up TLS on a cluster** section:
   - Remove the ordered list except for the first two steps
   - Create a bulleted list for the remaining configurations, as they don’t need to follow a specific order
- Add a note to clarify that `Encrypting HTTP Client Communications` is recommended for Kibana, Logstash and Beats.

Rationale:

During a Q&A session about Authentication and Authorisation in ES, it came out that support engineers couldn't help customers as the doc was confusing.

Preview: http://elasticsearch_48146.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/ssl-tls.html

### CODE REVIEW

**Comment by lcawl** [comment]
There are other reorganizations pending for this content.  Could you please refrain from merging these changes until we catch up?

**Comment by jkakavas** [comment]
I'll take a look at the actual proposed changes soon but a few comments:

- > During a Q&A session about Authentication and Authorisation in ES, it came out that support engineers couldn't help customers as the doc was confusing

   Can you share some notes from this? That would be very interesting for us

- we usually target master and then backport as needed, so we probably need to rebase this PR

**Comment by alaudazzi** [comment]
@jkakavas here is a short recap:

@corkscruff organized a basic security Q&A session within support and invited the only EMEA writer available (me) to address potential doc issues/improvements.
@pawel provided a great introduction on TLS, CA, private&public keys, encryption layers, etc. more in general on security, authorization and authentication. 
After the Q&A, Pawel and I met to discuss and agree on the doc changes proposed in this PR.

File: x-pack/docs/en/security/securing-communications/index.asciidoc [hunk 1]
```diff
@@ -8,6 +8,8 @@ manipulation of the data, and attempts to gain access to the server and thus the
 files storing the data. Securing your nodes helps reduce the risk from
 network-based attacks.
 
+WARNING: A full description of TLS/SSL, PKI (Public Key Infrastructure) certificates, and Certificate Authority is beyond the scope of this document. This page assumes prior knowledge of TLS/SSL as well as access to valid certificates.
```
Comment (jkakavas) [inline x-pack/docs/en/security/securing-communications/index.asciidoc]: I have found this kind of statements not helpful. I feel they are slightly intimidating and I'm not sure we should put them in the docs. I've even seen this type of statement ( we have this in other places, i.e. cloud saml/oidc setup ) being used in presentations as a testament to the difficulty and complexity of our docs and configuration. 

We aim to make the process as easy as possible so that even folks with minimum experience/knowledge can take a swing at configuring TLS (because they have no choice, TLS in transport is mandatory) and although the gist of the statement is true, stating it out loud doesn't buy us or the users much. Could we rephrase this to a note or a suggestion with a slightly more positive wording?

File: x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc [hunk 2]
```diff
@@ -19,21 +18,26 @@ applicable to your cluster.
 certificate and enable TLS on the transport layer. You can also optionally
 enable TLS on the HTTP layer. See
 {ref}/configuring-tls.html#tls-transport[Encrypting Communications Between Nodes in a Cluster] and
-{ref}/configuring-tls.html#tls-http[Encrypting HTTP Client Communications]. 
+{ref}/configuring-tls.html#tls-http[Encrypting HTTP Client Communications].
 
-. Configure the {monitor-features} to use encrypted connections. See <<secure-monitoring>>.
++
```
Comment (jkakavas) [inline x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc]: should we also hint to the fact that TLS on transport layer is mandatory, contrary to TLS on http being optional ?

File: x-pack/docs/en/security/securing-communications/index.asciidoc [hunk 3]
```diff
@@ -8,6 +8,8 @@ manipulation of the data, and attempts to gain access to the server and thus the
...
+
```
Comment (tvernum) [inline x-pack/docs/en/security/securing-communications/index.asciidoc]: This final part (`access to valid certificates`) doesn't seem right. The reader might want to use their own certificates, but we don't assume that they have them, as we provide options to create them.

File: x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc [hunk 4]
```diff
@@ -19,21 +18,26 @@ applicable to your cluster.
...
+NOTE: {ref}/configuring-tls.html#tls-http[Encrypting HTTP Client Communications] is strongly recommended if you want to enable TLS encryption for Kibana, Logstash and Beats.
```
Comment (jkakavas) [inline x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc]: "Enable TLS for X" might have multiple readings. It could mean "enable X to communicate over https to Elasticsearch", which is what we're trying to convey here I think, but it can also mean "Enable TLS on X itself so that other components can talk to X over TLS" , i.e. browser -> kibana, beats -> logstash etc. 

I think our stance should be that enabling TLS on http is strongly recommended unless there is a very specific reason not to. If we were to frame it as a conditional here, I think we should mention something like : 

```suggestion
NOTE: If you want to enable other stack components like Kibana, Logstash and Beats , or your users via their browsers to communicate securely with Elasticsearch, then {ref}/configuring-tls.html#tls-http[TLS configuration for HTTP] is strongly required so that the communication is encrypted with TLS. 
```

**Review by jkakavas — COMMENTED** [review]
Thanks for this @alaudazzi, I left some comments !

File: x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc (hunk 4 above)
Comment (immon) [inline x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc]: Encrypting HTTP Client Communications is **required** if you want to enable TLS ....

File: x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc [hunk 5]
```diff
@@ -9,8 +9,7 @@ WARNING: Clusters that do not have encryption enabled send all data in plain tex
 including passwords. If the {es} {security-features} are enabled, unless you have a trial license, you must configure SSL/TLS for internode-communication.
 
 The following steps describe how to enable encryption across the various
-components of the {stack}. You must perform each of the steps that are
-applicable to your cluster.
```
Comment (immon) [inline x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc]: not sure why we removed the last sentence.

**Review by immon — CHANGES_REQUESTED** [review]

File: x-pack/docs/en/security/securing-communications/index.asciidoc (hunk 1 above)
Comment (tvernum) [reply x-pack/docs/en/security/securing-communications/index.asciidoc]: One option is for us to actually write that _Introduction to TLS_ documentation and link to it here.
I don't expecially want to explain TLS to users, but I'd rather do that then say something to the effect of _If you don't already know this information then Elasticsearch isn't for you_.

Note, this isn't just true for TLS. We also have that issue with things like YAML (for config files) so it might be worth a separate conversation around how do we help new users learn the prereq knowledge to operate ES well.

Comment (debadair) [reply x-pack/docs/en/security/securing-communications/index.asciidoc]: If we're not providing enough information for people to successfully secure the stack, that's a bug IMO. I think providing an overview of TLS in the context of the Elastic Stack would be worthwhile. We don't have to go into the details of *how* public key encryption works, but we could definitely do a better job of introducing what's required and how it all fits together. 

There have been a lot of improvements around security to make things easier, @lcawl added some tutorials to make it clearer what you need to do, and there's recently been a lot of reshuffling of this content. But some of the incremental changes that were made along the way made sense at the time, but haven't resulted in a very coherent story. I think it's due for more than minor cleanup.

File: x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc (hunk 4 above)
Comment (debadair) [reply x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc]: I'd make this a stronger statement--completely configuring security should be the recommended path. Something like:

> To enable Kibana, Logstash, and Beats to communicate with Elasticsearch securely, you must configure TLS for HTTP. 

It's not just a suggestion, it's a requirement.

File: x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc [hunk 6]
```diff
@@ -9,8 +9,7 @@ WARNING: Clusters that do not have encryption enabled send all data in plain tex
...
+components of the {stack}.
```
Comment (debadair) [inline x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc]: As the person responsible for introducing the "encrypting communications" language, I think we should instead be talking about configuring TLS. Something like:

> To configure TLS to encrypt inter-node communications on the transport layer and enable stack components to connect to Elasticsearch over HTTPS, you need to:

File: x-pack/docs/en/security/securing-communications/index.asciidoc (hunk 1 above)
Comment (lcawl) [reply x-pack/docs/en/security/securing-communications/index.asciidoc]: I will agree that this note doesn't solve the bigger problem, so I'm removing it from this PR.

**Comment by lcawl** [comment]
@elasticmachine run elasticsearch-ci/docs

File: x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc (hunk 5 above)
Comment (lcawl) [reply x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc]: I don't really think this introduction is necessary anymore, so I've removed the whole thing

File: x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc (hunk 4 above)
Comment (lcawl) [reply x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc]: I've implemented a combination of these suggestions.

**Comment by lcawl** [comment]
>   Can you share some notes from this? That would be very interesting for us

I agree that it would be helpful to know where folks are encountering problems, since I think most of these suggestions are quite superficial and won't solve major problems. I didn't have problems when I followed the steps in the tutorial, so there must be a level of complexity in real customer environments that we're just not addressing. 

> * we usually target master and then backport as needed, so we probably need to rebase this PR

I've rebased this PR on master.

**Comment by lcawl** [comment]
>   Can you share some notes from this? That would be very interesting for us

I agree that it would be helpful to know where folks are encountering problems, since I think most of these suggestions are quite superficial and won't solve major problems. I didn't have problems when I followed the steps in the tutorial, so there must be a level of complexity in real customer environments that we're just not addressing. 

> * we usually target master and then backport as needed, so we probably need to rebase this PR

I've rebased this PR on master.

**Comment by lcawl** [comment]
> * Move the **Encrypting communications** section after the Overview.

I reverted this change when I rebased the PR.  I don't think this move solves any of the larger problems.  I also don't really think of the sections in the table of contents as sequential (i.e. some users might research authentication and authorization before reading about encryption and I don't think either of those paths are wrong).  If folks are walking a sequential walk-through, I think the tutorials are a better source for that.

**Review by lcawl — APPROVED** [review]
I've addressed merge conflicts in this PR and though it's smaller changes than originally proposed, I think it's nonetheless still worthwhile. LGTM

File: x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc [hunk 7]
```diff
@@ -6,32 +6,37 @@ within your {es} cluster. Connections are secured using Transport Layer Security
...
+NOTE: To enable {kib}, {ls}, and Beats to communicate securely with {es}, you
+must configure TLS for HTTP. Likewise, it enables your users to communicate
+securely with {es} via their browsers.
+
+--
+
+Now you can proceed with the configurations that are applicable to your cluster:
 
-. Configure the {monitor-features} to use encrypted connections. See <<secure-monitoring>>.
+* Configure the {monitor-features} to use encrypted connections. See <<secure-monitoring>>.
```
Comment (nrichers) [inline x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc:29]: Can these be performed in any order? If not, the ordered steps might make more sense.

**Review by nrichers — APPROVED** [review]
LGTM. One minor question, but feel free to ignore if you don't think it's relevant.

File: x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc (hunk 7 above)
Comment (alaudazzi) [reply x-pack/docs/en/security/securing-communications/setting-up-ssl.asciidoc:29]: I would be tempted to say no, but I'll leave to @lcawl to confirm.
